test/cpp/tensorexpr/tutorial.cpp

```diff
@@ -0,0 +1,426 @@
+        });
+    std::cout << "Tensor computation: " << *R << std::endl;
+    // Prints: Tensor computation: Tensor R(i[64], j[32]) = Z(i, j) * P[i, j]
+
+    // Placeholders could be thought of as external tensors, i.e. tensors for
+    // which we don't have the element expression. In other words, for `Tensor`
+    // we know an expression specifying how its elements can be computed (a
+    // mathematical formula). For external tensors, or placeholders, we don't
+    // have such an expression. They need to be considered as coming to us as
+    // inputs from outside - we can only load data from them.
+    //
+    // Also note that we use 'call' to construct an access to an element of a
+    // Tensor and we use 'load' for accessing elements of an external tensor
+    // through its Placeholder. This is an implementation detail and could be
+    // changed in future.
+    //
+    // Why do we have Functions and Tensors and what is the relationship between
+    // them? Functions are used to represent several computations performed over
+    // the same domain. Tensors refer to individual computations of a Function.
+    //
+    // Also note that currently a lot of code only supports single-output
+    // Functions, in which case they become almost identical to Tensors. This
+    // probably will be changed in future.
+
+    // TODO: Show how reductions are represented and constructed
+  }
+
+  std::cout << "*** Loopnests and Statements ***" << std::endl;
+  {
+    // Creating a tensor expression is the first step to generate an executable
+    // code for it. A next step is to represent it as a loop nest and apply
+    // various loop transformations in order to get an optimal implementation.
+    // In Halide's or TVM's terms the first step was to define the algorithm of
+    // computation (what to compute?) and now we are getting to the schedule of
+    // the computation (how to compute?).
+    //
+    // Let's create a simple tensor expression and construct a loop nest for it.
+    Placeholder A("A", kFloat, {64, 32});
+    Placeholder B("B", kFloat, {64, 32});
+    Tensor* X = Compute(
+        "X",
+        {{64, "i"}, {32, "j"}},
+        [&](const VarHandle& i, const VarHandle& j) {
+          return A.load(i, j) + B.load(i, j);
+        });
+    Tensor* Y = Compute(
+        "Y",
+        {{64, "i"}, {32, "j"}},
+        [&](const VarHandle& i, const VarHandle& j) {
+          return sigmoid(X->call(i, j));
+        });
+    std::cout << "Tensor computation X: " << *X
+              << "Tensor computation Y: " << *Y << std::endl;
+    // Prints:
+    // Tensor computation X: Tensor X(i[64], j[32]) = (A[i, j]) + (B[i, j])
+    // Tensor computation Y: Tensor Y(i[64], j[32]) = sigmoid(X(i, j))
+
+    // Creating a loop nest is as quite simple, we just need to specify what are
+    // the output tensors in our computation and LoopNest object will
+    // automatically pull all tensor dependencies:
```

Comment:
Would the reader understand what "pull[ing] all tensor dependencies" means or why it's important?